packages/stockflux-watchlist/src/components/watchlist/Watchlist.js

```diff
@@ -24,6 +23,10 @@ const getDistinctElementArray = array => [...new Set(array)];
 const Watchlist = () => {
   const [dragOverIndex, setDragOverIndex] = useState(null);
   const [unwatchedSymbol, setUnwatchedSymbol] = useState(null);
+  const [displayPreview, setDisplayPreview] = useState(false);
+  const [previewOptions, setPreviewOptions] = useState();
+  const [previewPosition, setPreviewPosition] = useState({ top: 0, left: 0 });
+  const [previewSize, setPreviewSize] = useState({ height: 300, width: 300 });
```

Comment:
Just a suggestion - maybe it would better to hold all preview state in a single `useState` object? 
```
const [preview, setPreview] = useState(
  {
    options: null,
    position: { top: 0, left: 0 },
    size: { height: 300, width: 300 },
  }
);
```

And the use it with object spread like this:

`setPreview({...preview, size: { height: 500, width: 500 }})`

Reply:
combined the  `previewOptions`, `previewPosition` and `previewSize` together into `previewDetails`. Added `// eslint-disable-next-line react-hooks/exhaustive-deps` to the hooks, but they are correct as I only want them to first on specific values changing.